src/backend/distributed/utils/citus_depended_object.c

```diff
@@ -285,3 +287,184 @@ GetCitusDependedObjectArgs(int pgMetaTableVarno, int pgMetaTableOid)
 
 	return list_make2((Node *) metaTableOidConst, (Node *) oidVar);
 }
+
+
+/*
+ * ShouldValidateStmt decides if the given statement is amongst
+ * the statements which causes pg tests to fail.
+ */
+static bool
+ShouldValidateStmt(Node *node)
```

Comment:
why do we ever bother filtering for specific ops? I was more considering something
```C

ops = GetDistributeObjectOps(parsetree);
if (!ops) return false; /* unrelated to citus, do not interfere*/

if (ops->address)
{
List *objectAddresses = ops->address(node, true);
ObjectAddress *objectAddress = NULL;
foreach_ptr(objectAddress, objectAddresses)
{
    if (!Valid) return false; /* this is an object with invalid address, let PG handle, Citus cannot do anything meaningful anyway */
}}
```

Reply:
There were 2 bugs in address methods for reindex and foreign server statements, which cause related tests to exit with status 2. That is why I filtered statements. But, those bugs are resolved by myself, and now I safely removed ShouldValidateStmt.